Add/remove some params for encoder sample

LiuChiachi opened the pull request:
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Function optimization
### PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->
APIs
### Description
<!-- Describe what this PR does -->
- remove and add params for `SemanticIndexingPredictor`
- Add warning for `activation` except  `relu` and `gelu`.

guoshengCS commented:
bert_encoder_transformer.h 这个文件在 #974 中应该会被整个替换，估计当前这种在原文件上sed替换的可能后面会有些冲突会被覆盖，不过`$<SEMICOLON>`这个用法还挺好的， @tianxin1860 可以再确认下

LiuChiachi committed:
recover bert_encoder_transformer.h

remove useless parameters, and remove redefination of pretrain model, and update layer replacement for considering each layer independently.

LiuChiachi commented:
> bert_encoder_transformer.h 这个文件在 #974 中应该会被整个替换，估计当前这种在原文件上sed替换的可能后面会有些冲突会被覆盖，不过`$<SEMICOLON>`这个用法还挺好的， @tianxin1860 可以再确认下

已经把`bert_encoder_transformer.h`这块的更改删除了。@tianxin1860 可以不需要关注了:)

guoshengCS commented on examples/semantic_indexing/faster_predict.py:
```diff
@@ -97,7 +96,12 @@ def __init__(self,
+            dropout,
+            activation,
+            bias_attr=bias_attr)
         self.ptm.encoder = TransformerEncoder(encoder_layer, n_layer)
```
这里为什么要把预训练模型的ptm.encoder换成新创建的TransformerEncoder呢

LiuChiachi replied:
感谢指出，已经删除了不需要的参数

guoshengCS commented on examples/semantic_indexing/faster_predict.py:
```diff
@@ -83,8 +83,7 @@ def __init__(self,
                  dropout=0,
-                 max_seq_len=128,
-                 is_gelu=False):
+                 bias_attr=None):
```
这些参数应该去掉，pretrained_model传进来了这些参数应该就是由pretrained_model控制从pretrained_model获取的

LiuChiachi replied:
感谢指出，已经删除了不需要的参数

tianxin1860 commented on examples/semantic_indexing/faster_predict.py:
```diff
@@ -97,7 +96,12 @@ def __init__(self,
+            dropout,
+            activation,
+            bias_attr=bias_attr)
         self.ptm.encoder = TransformerEncoder(encoder_layer, n_layer)
```
为什么需要替换 `self.ptm.encoder` 而不使用原有的 `self.ptm.encoder` ?

LiuChiachi replied:
感谢指出，已经删除了不需要的参数

tianxin1860 reviewed (commented):
Leave a comment

guoshengCS commented on paddlenlp/ops/faster_transformer/transformer/encoder.py:
```diff
@@ -251,14 +251,17 @@ def enable_faster_encoder(self):
 
     def check_if_usable(layer):
```
感觉apply和check的设计不太完善，check只check layer的直接childern中的TransformerEncoderLayer，apply是所有layer，后面可以在apply里check

LiuChiachi replied:
感谢指出，现在已对每个layer单独考虑，判断和替换写在一个func里，借助apply完成递归。

LiuChiachi committed:
Merge branch 'develop' into update-encoder-patch

guoshengCS commented on paddlenlp/ops/faster_transformer/transformer/encoder.py:
```diff
@@ -252,35 +251,34 @@ def enable_faster_encoder(self):
                 "Exception occurs when using Faster Transformer. " \
                 "The original forward will be involved. ")
+            return self
+        for layer in self.children():
```
这里最好是layer自己apply就可以了，比如layer本身是TransformerEncoder

ZeyuChen committed:
Merge branch 'develop' into update-encoder-patch
